include/javabind/core.hpp

```diff
@@ -721,12 +730,15 @@ namespace javabind
     template <> struct ArgType<bool> { using type = JavaBooleanType; };
     template <> struct ArgType<int8_t> { using type = JavaByteType; };
     template <> struct ArgType<char16_t> { using type = JavaCharacterType; };
-    template <> struct ArgType<int16_t> { using type = JavaShortType; };
-    template <> struct ArgType<int32_t> { using type = JavaIntegerType; };
-    template <> struct ArgType<int64_t> { using type = JavaLongType; };
+    template <> struct ArgType<int16_t> { using type = JavaShortType<int16_t>; };
+    template <> struct ArgType<int32_t> { using type = JavaIntegerType<int32_t>; };
+    template <> struct ArgType<int64_t> { using type = JavaLongType<int64_t>; };
+    template <> struct ArgType<uint8_t> { using type = JavaShortType<uint8_t>; };
+    template <> struct ArgType<uint16_t> { using type = JavaIntegerType<uint16_t>; };
+    template <> struct ArgType<uint32_t> { using type = JavaLongType<uint32_t>; };
```

Comment:
These three would be included only if `JAVABIND_INTEGER_WIDENING_CONVERSION` is defined.